### PR TITLE
#164797982 Post comment

### PR DESCRIPTION
**Pull request by McHardex**
#### Description
This PR ensures that a User can  comment on an article
#### Type of change
- [x] New feature
- [x]

#### How Has This Been Tested?
- [x] Unit testing

#### PT-ID

[164797982](https://www.pivotaltracker.com/story/show/164797982)

### CODE REVIEW

**Review by sojida — APPROVED** [review]

**Review by walsamlee — APPROVED** [review]

File: src/components/presentations/Article/ArticlePage.jsx [hunk 1]
```diff
@@ -12,18 +14,22 @@ class ArticlePage extends Component {
     this.state = {};
   }
 
-  componentDidMount = () => {
+  componentDidMount = async () => {
```
Comment (rotimi-babalola) [inline src/components/presentations/Article/ArticlePage.jsx]:
```
async componentDidMount {
   ....
}
```

File: src/components/presentations/Article/ArticlePage.jsx [hunk 2]
```diff
@@ -12,18 +14,22 @@ class ArticlePage extends Component {
...
     const { articleId } = match.params;
-    getSingleArticle(articleId);
+    await getSingleArticle(articleId);
   };
 
   render() {
-    const { singleArticle } = this.props;
-    const { article } = singleArticle;
+    const { singleArticle, match, postComment, isLoadingReducer } = this.props;
+    const { loader } = isLoadingReducer;
```
Comment (rotimi-babalola) [inline src/components/presentations/Article/ArticlePage.jsx]: ```const { loader: showLoader } = isLoadingReducer```

File: src/components/presentations/Article/ArticlePage.jsx [hunk 3]
```diff
@@ -42,7 +48,23 @@ class ArticlePage extends Component {
                     </div>
                   </Grid.Column>
                   <Grid.Column>
-                    <Comments comments={article.Comments} />
+                    <h3>Comments</h3>
+                    {article && article.author && (
```
Comment (rotimi-babalola) [inline src/components/presentations/Article/ArticlePage.jsx]: To check that an object is empty we can use `isEmpty` lodash - `isEmpty(obj)`

Alternatively,

Create a util function that checks that the object does not have any keys to ensure that it is empty

File: src/components/presentations/Article/ArticlePage.jsx [hunk 4]
```diff
@@ -42,7 +48,23 @@ class ArticlePage extends Component {
...
+                      <InputComment
+                        imageUrl={article.author.image_url}
+                        articleId={article.id}
+                        postComment={postComment}
+                      />
+                    )}
+                    {comments &&
```
Comment (rotimi-babalola) [inline src/components/presentations/Article/ArticlePage.jsx]: Create and use a util function to check that an array is empty here

**Review by vic3king — COMMENTED** [review]
please fix conflicts so we can merge this.

**Review by vic3king — APPROVED** [review]

**Pull request merged**